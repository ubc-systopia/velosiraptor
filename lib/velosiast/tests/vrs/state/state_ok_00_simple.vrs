state() {
    reg foo [ 8 ]
}

state(base: addr) {
    mem foo [ base, 8, 8 ]
}

// two fields
state(base: addr) {
    mem foo [ base, 8, 8 ],
    reg bar [ 8 ]
}

state(base: addr) {
    reg foo [ 8 ],
    reg bar [ 8 ]
}

state(base: addr) {
    mem foo [ base, 0, 8 ],
    mem bar [ base, 8, 8 ],
}

// with slices

state() {
    reg foo [ 8 ] {
        0..64 val
    }
}

state(base: addr) {
    reg foo [ 8 ] {
        0..32 val,
        32 .. 64 val2
    }
}

state() {
    mem foo [ base, 8, 8 ] {
        0..64 val
    }
}

state(base: addr) {
     mem foo [ base, 8, 8 ] {
        0..32 val,
        32 .. 64 val2
    }
}

state(base: addr) {
     mem foo [ base, 8, 8 ] {
        0..32 val,
        32 .. 64 val2,
    }
}

state(base: addr) {
     mem foo [ base, 0, 8 ] {
        0..32 val,
        32 .. 64 val2
    },
     mem bar [ base, 8, 8 ] {
        0..32 val,
        32 .. 64 val2
    }
}