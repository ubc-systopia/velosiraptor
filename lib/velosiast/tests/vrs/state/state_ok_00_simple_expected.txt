StateDef() {
    reg state.foo [8] {
      0..64 state.foo.val,
    },
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

StateDef(base: addr) {
    mem state.foo [base, 8, 8] {
      0..64 state.foo.val,
    },
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

StateDef(base: addr) {
    mem state.foo [base, 8, 8] {
      0..64 state.foo.val,
    },
    reg state.bar [8] {
      0..64 state.bar.val,
    },
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

StateDef(base: addr) {
    reg state.foo [8] {
      0..64 state.foo.val,
    },
    reg state.bar [8] {
      0..64 state.bar.val,
    },
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

StateDef(base: addr) {
    mem state.foo [base, 0, 8] {
      0..64 state.foo.val,
    },
    mem state.bar [base, 8, 8] {
      0..64 state.bar.val,
    },
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

StateDef() {
    reg state.foo [8] {
      0..64 state.foo.val,
    },
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

StateDef(base: addr) {
    reg state.foo [8] {
      0..32 state.foo.val,
      32..64 state.foo.val2,
    },
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

StateDef() {
    mem state.foo [base, 8, 8] {
      0..64 state.foo.val,
    },
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

StateDef(base: addr) {
    mem state.foo [base, 8, 8] {
      0..32 state.foo.val,
      32..64 state.foo.val2,
    },
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

StateDef(base: addr) {
    mem state.foo [base, 0, 8] {
      0..32 state.foo.val,
      32..64 state.foo.val2,
    },
    mem state.bar [base, 8, 8] {
      0..32 state.bar.val,
      32..64 state.bar.val2,
    },
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

